### PR TITLE
programs.neovim: add extraLuaConfig

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
### Description

Suggestion to fix: https://github.com/nix-community/home-manager/issues/3457

Adds a configuration `programs.neovim.extraLuaConfig` that is added to the `init.lua` generated by home-manager.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/neovim.nix [hunk 1]
```diff
@@ -394,8 +405,9 @@ in {
               "vim.cmd [[source ${
                 pkgs.writeText "nvim-init-home-manager.vim"
                 neovimConfig.neovimRcContent
-              }]]" + lib.optionalString hasLuaConfig
-              config.programs.neovim.generatedConfigs.lua;
+              }]]" + config.programs.neovim.extraLuaConfig
+              + (lib.optionalString hasLuaConfig
+                config.programs.neovim.generatedConfigs.lua);
```
Comment (ncfavier) [inline modules/programs/neovim.nix:410]:
```suggestion
              + lib.optionalString hasLuaConfig
                config.programs.neovim.generatedConfigs.lua;
```

File: tests/modules/programs/neovim/extra-lua-init.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    programs.neovim = {
+      enable = true;
+
+      extraLuaConfig = ''
+        -- extraLuaConfig
+      '';
+    };
+    nmt.script = ''
+      nvimFolder="home-files/.config/nvim"
+        assertFileContent "$nvimFolder/init.lua" ${
```
Comment (ncfavier) [inline tests/modules/programs/neovim/extra-lua-init.nix:16]: Indentation is off here

File: modules/programs/neovim.nix (hunk 1 above)
Comment (GuillaumeDesforges) [reply modules/programs/neovim.nix]: Addressed in e12229

File: tests/modules/programs/neovim/extra-lua-init.nix (hunk 2 above)
Comment (GuillaumeDesforges) [reply tests/modules/programs/neovim/extra-lua-init.nix]: Addressed in e12229

**Review by ncfavier — APPROVED** [review]

File: tests/modules/programs/neovim/extra-lua-init.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      enable = true;
+
+      extraLuaConfig = ''
+        -- extraLuaConfig
+      '';
+    };
+    nmt.script = ''
+      nvimFolder="home-files/.config/nvim"
+      assertFileContent "$nvimFolder/init.lua" ${
+        pkgs.writeText "init.lua-expected" "-- extraLuaConfig "
```
Comment (ncfavier) [inline tests/modules/programs/neovim/extra-lua-init.nix:17]:
```suggestion
        pkgs.writeText "init.lua-expected" ''
          -- extraLuaConfig
        '';
```

Comment (GuillaumeDesforges) [reply tests/modules/programs/neovim/extra-lua-init.nix:17]: Thanks for spotting that, I got ahead of myself 😅

Comment (GuillaumeDesforges) [reply tests/modules/programs/neovim/extra-lua-init.nix]: Fixed in 6ee2f04

**Pull request merged**